core/templates/dev/head/domain/collection/CollectionValidationService.js

```diff
@@ -72,6 +72,28 @@ oppia.factory('CollectionValidationService', [
       });
     };
 
+    // Validates that the tags for the collection are in the proper format,
+    // returns true if all tags are in the correct format.
+    var validateTagFormat = function(tags) {
+      // Check to ensure that all tags follow the format specified in
+      // TAG_REGEX.
+      for (var i = 0; i < tags.length; i++) {
+        var tagRegex = new RegExp(GLOBALS.TAG_REGEX);
+        if (!tags[i].match(tagRegex)) {
+          return false;
+        }
+      }
+      return true;
+    };
+
+    // Validates that the tags for the collection do not have duplicates,
+    // returns true if there are duplicates.
+    var validateDuplicateTags = function(tags) {
+      return tags.every(function(tag, idx) {
+        return tags.indexOf(tag, idx + 1) === -1;
+      });
+    };
+
```

Comment:
Would it be possible to validate that the tags are indeed normalized (i.e. doesn't contain more than one consecutive whitespace character)? One way to do this could be to check that applying normalization to the tag doesn't change it.